### DISCUSSION TITLE
Certain font awesome glyphs not working despite importing full set

### DISCUSSION THREAD

**Discussion by Monklite** [razonyang/hugo-theme-bootstrap#554] [General]
In `params.toml`, I have the following:

```
# Custom CSS and JS
customCSS = [
]
customJS = [
	"assets/icons/custom.js"
]
```

`assets/icons/custom.js` contains the following:

```
customJS = [
  "https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.4/js/all.min.js" // Full icons set.
]
```

Despite this, glyphs in the menu for font awesome are not loading; for example, `menu.toml` contains:

```
[[main]]
  id = "projects"
  parent = "dropdown"
  name = "Projects"
  url = "/projects"
  pre = '<i class="fas fa-fw fa-diagram-project"></i>'
  weight = 10
```

and it just loads some flashing error icon. Changing `fas` to `fab` does the same. Importing the full icon set was supposed to fix this and others seem to have luck when doing this based on what I have seen, I'm pretty sure I am doing something incorrectly. Any insight?

Hugo version is 0.99.1 (Using extended from the AUR) on Arch Linux. Also worth noting I used the skeleton version.

**Comment by razonyang** [comment]
Hi, I think you misunderstand the `assets/icons/custom.js`, it's no need to add `customJS` in `params.toml`, since the `custom.js` file will be compiled into the icons' bundle automatically.

And the `assets/icons/custom.js` should be something like the following, it's a JS file instead of a configuration file:

```js
import { faProjectDiagram } from '@fortawesome/free-solid-svg-icons';
// import { faAddressBook } from '@fortawesome/free-regular-svg-icons';
// import { faAmazon, faGoogle } from '@fortawesome/free-brands-svg-icons';

const icons = [
    faProjectDiagram, 
    // faAddressBook,
    // faAmazon, faGoogle,
];
export default icons;
```

HTML: `<i class="fas fa-project-diagram"></i>` instead of `<i class="fa-solid fa-diagram-project"></i>`.

Because the theme uses FontAwesome` v5` not `v6`.

> In this way, you can load icons that you need instead of loading a large icons set.

See also:

- https://hbs.razonyang.com/en/docs/look-and-feel/#icons
- https://github.com/razonyang/hugo-theme-bootstrap/blob/develop/exampleSite/assets/icons/custom.js

**Reply by Monklite** [reply]
> Hi, I think you misunderstand the `assets/icons/custom.js`, it's no need to add `customJS` in `params.toml`, since the `custom.js` file will be complied into the icons' bundle **automatically.
> 
> And the `assets/icons/custom.js` should be something like the following, it's a JS file instead of a configuration file:
> 
> ```js
> import { faProjectDiagram } from '@fortawesome/free-solid-svg-icons';
> // import { faAddressBook } from '@fortawesome/free-regular-svg-icons';
> // import { faAmazon, faGoogle } from '@fortawesome/free-brands-svg-icons';
> 
> const icons = [
>     faProjectDiagram, 
>     // faAddressBook,
>     // faAmazon, faGoogle,
> ];
> export default icons;
> ```
> 
> HTML: `<i class="fas fa-project-diagram"></i>` instead of `<i class="fa-solid fa-diagram-project"></i>`.
> 
> Because the theme uses FontAwesome` v5` not `v6`.
> 
> > In this way, you can load icons that you need instead of loading a large icons set.
> 
> See also:
> 
>     * https://hbs.razonyang.com/en/docs/look-and-feel/#icons
> 
>     * https://github.com/razonyang/hugo-theme-bootstrap/blob/develop/exampleSite/assets/icons/custom.js

It worked. Thank you so much, that makes a lot more sense.

**Reply by razonyang** [reply]
My pleasure.

**Comment by razonyang** [comment]
You can also load full icons' set by setting `customJS` in params.toml:

```
customJS = [
  "https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.4/js/all.min.js" // Full icons set.
]
```
